scripts/globals/nyzul/floor_generation.lua

```diff
@@ -0,0 +1,1245 @@
+-----------------------------------
+-- Nyzul Isle: Floor generation methods and data.
+-----------------------------------
+local ID = zones[xi.zone.NYZUL_ISLE]
+-----------------------------------
+xi = xi or {}
+xi.nyzul = xi.nyzul or {}
+
+-----------------------------------
+-- Data Tables
+-----------------------------------
+local layoutSpawnPoints = -- Spawnpoints by layout.
+{
+    [1] =
+    {
+        [ 1] = { x = 405.5, y = 0, z = -512.5 },
+        [ 2] = { x = 406.0, y = 0, z = -487.0 },
+        [ 3] = { x = 414.5, y = 0, z = -491.2 },
+        [ 4] = { x = 414.9, y = 0, z = -504.1 },
+        [ 5] = { x = 424.5, y = 0, z = -503.9 },
+        [ 6] = { x = 424.5, y = 0, z = -491.7 },
+        [ 7] = { x = 433.2, y = 0, z = -486.6 },
+        [ 8] = { x = 433.1, y = 0, z = -499.1 },
+        [ 9] = { x = 433.8, y = 0, z = -513.0 },
+        [10] = { x = 420.5, y = 0, z = -513.1 },
+        [11] = { x = 420.0, y = 0, z = -526.6 },
+        [12] = { x = 429.0, y = 0, z = -530.8 },
+        [13] = { x = 428.5, y = 0, z = -548.7 },
+        [14] = { x = 420.0, y = 0, z = -549.1 },
+        [15] = { x = 410.9, y = 0, z = -549.0 },
+        [16] = { x = 410.9, y = 0, z = -540.2 },
+        [17] = { x = 420.0, y = 0, z = -539.6 },
+        [18] = { x = 380.4, y = 0, z = -526.0 },
+        [19] = { x = 367.0, y = 0, z = -526.9 },
+        [20] = { x = 375.0, y = 0, z = -536.0 },
+        [21] = { x = 385.3, y = 0, z = -540.0 },
+        [22] = { x = 375.8, y = 0, z = -544.0 },
+        [23] = { x = 366.8, y = 0, z = -553.5 },
+        [24] = { x = 379.1, y = 0, z = -553.4 },
+        [25] = { x = 393.0, y = 0, z = -554.0 },
+        [26] = { x = 393.7, y = 0, z = -541.0 },
+        [27] = { x = 393.5, y = 0, z = -526.5 },
+        [28] = { x = 379.8, y = 0, z = -566.0 },
+        [29] = { x = 380.0, y = 0, z = -579.8 },
+        [30] = { x = 393.3, y = 0, z = -579.6 },
+        [31] = { x = 473.4, y = 0, z = -593.2 },
+        [32] = { x = 473.0, y = 0, z = -580.0 },
+        [33] = { x = 473.4, y = 0, z = -567.0 },
+        [34] = { x = 464.6, y = 0, z = -575.9 },
+        [35] = { x = 459.8, y = 0, z = -585.0 },
+        [36] = { x = 456.0, y = 0, z = -575.7 },
+        [37] = { x = 446.4, y = 0, z = -566.2 },
+        [38] = { x = 446.7, y = 0, z = -579.6 },
+        [39] = { x = 446.4, y = 0, z = -593.4 },
+        [40] = { x = 459.9, y = 0, z = -594.0 },
+        [41] = { x = 433.8, y = 0, z = -580.1 },
+        [42] = { x = 433.5, y = 0, z = -567.1 },
+        [43] = { x = 420.5, y = 0, z = -566.2 },
+        [44] = { x = 406.5, y = 0, z = -566.7 },
+        [45] = { x = 406.5, y = 0, z = -579.7 },
+        [46] = { x = 406.5, y = 0, z = -593.4 },
+        [47] = { x = 415.5, y = 0, z = -584.3 },
+        [48] = { x = 419.9, y = 0, z = -575.5 },
+        [49] = { x = 424.7, y = 0, z = -584.4 },
+        [50] = { x = 433.2, y = 0, z = -593.2 },
+        [51] = { x = 405.5, y = 0, z = -499.8 },
+    },
+
+    [2] =
+    {
```

Comment:
is there anyway we can make this less of a sea of numbers? I know it was one before you got here, but I can't brain.

Reply:
its a table of possitions based on a layout. I COULD enum the layouts, like SOUTH_WEST, NORTH, etc... but the actual positions cant be helped until we have a system that allows to define a region and select valid positions from said region

Reply:
I did manage to remove one full column (rotation) from the positions, couse it was always math.random(1, 255)

Reply:
maybe comment what the keys are rather than enuming directions. anything to break it up visually and break it down for the reader. like imagine the intem_mods table with zero commenting. that's what this feels like.

Reply:
Sure, I can do that